common:
    lid_topic:  "/iv_points"
    gps_topic: "/ublox_driver/receiver_pvt"
    imu_topic:  "/livox/imu"
    uwb_topic: "/nlink_linktrack_nodeframe2"
    vicon_topic: "/vrpn_client_node/uwb_ngx_tag/pose"
    file_save_path: "/home/mint/ws_fusion_uwb/src/igo_eskf/data/res/"
    en_vicon: true
    en_debug: true
    time_sync_en: false         # ONLY turn on when external time synchronization is really not possible
    time_offset_lidar_to_imu: 0.0 # Time offset between lidar and IMU calibrated by other algorithms, e.g. LI-Init (can be found in README).
                                  # This param will take effect no matter what time_sync_en is. So if the time offset is not known exactly, please set as 0.0

covariance:
    noise:
        posi: [1.0e-6, 1.0e-6, 1.0e-6]
        vel: [1.0e-4, 1.0e-4, 1.0e-4]
        ori: [1.0e-6, 1.0e-6, 1.0e-6]
        gyro: [1.0e-6, 1.0e-6, 1.0e-6]
        acc: [1.0e-6, 1.0e-6, 1.0e-6]
    process:
        gyro: [1.0e-3, 1.0e-3, 1.0e-3]
        accel: [1.0e-3, 1.0e-3, 1.0e-3]
    measurement:
        posi: [1.0e-3, 1.0e-3, 1.0e-3]
        vel: [1.0e-3, 1.0e-3, 1.0e-3]



        
mapping:
    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    fov_degree:    120
    det_range:     450.0
    extrinsic_est_en:  false      # true: enable the online estimation of IMU-LiDAR extrinsic 是否将lidar-IMU外参当作状态进行优化
#   extrinsic_T: [ 1.6974312355289280963, -0.018917936385979094428, -1.502743887910872071 ]
    extrinsic_T: [ 0.0, 0.0, 0.0 ]
    extrinsic_R: [ 1.0, 0.0, 0.0,
                                0.0, 1.0, 0.0,
                                0.0, 0.0, 1.0]


publish:
    path_en:  true
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: true       # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: true  # true: output the point cloud scans in IMU-body-frame

pcd_save:
    pcd_save_en: false
    interval: -1                 # how many LiDAR frames saved in each pcd file; 
                                 # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.
